### PR TITLE
Jhe 258151 playground mock

### PR DESCRIPTION
**Pull request by jhe-okta**
- have SIW able to recognize remediation 'device-challenge-poll' and render DeviceChallengePoll view
- in DeviceChallengePoll view, do polling and probing

Not included:
- new UX
- test

### CODE REVIEW

File: src/v2/view-builder/views/DeviceChallengePollView.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      xhr.setRequestHeader('Content-Type', 'application/json');
+    }
+    xhr.send(body);
+  });
+};
+
+const Body = BaseForm.extend({
+  noButtonBar: true,
+  initialize () {
+    this.deviceChallengePollRemediation = this.options.appState.get('currentState').remediation.find(
```
Comment (haishengwu-okta) [inline src/v2/view-builder/views/DeviceChallengePollView.js]: could you add function like `getRemediationForm` to appstate ?

File: src/v2/view-builder/views/DeviceChallengePollView.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      rem => rem.name === 'device-challenge-poll'
+    );
+    this.doChallenge();
+    BaseForm.prototype.initialize.apply(this, arguments);
```
Comment (haishengwu-okta) [inline src/v2/view-builder/views/DeviceChallengePollView.js]: call super shall happen before other statements.

File: src/v2/view-builder/views/DeviceChallengePollView.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+  stopPolling () {
+    if (this.polling) {
+      console.log('stop polling');
+      clearInterval(this.polling);
+    }
+  },
+});
+
+const Footer = BaseFooter.extend({
```
Comment (haishengwu-okta) [inline src/v2/view-builder/views/DeviceChallengePollView.js]: looks it's identical to Identify.footer?
we could do
```
import IdentifierView from './views/IdentifierView';
const Footer = Identifier.prototype.Footer;
```

File: src/v2/view-builder/views/DeviceChallengePollView.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
+/* global Promise */
+import { _ } from 'okta';
+import BaseView from '../internals/BaseView';
+import BaseForm from '../internals/BaseForm';
+import BaseFooter from '../internals/BaseFooter';
+
+const request = (url, method = 'GET', body = null, timeout = 1000) => {
```
Comment (haishengwu-okta) [inline src/v2/view-builder/views/DeviceChallengePollView.js]: why not use jquery.ajax or  'cross-fetch';

File: src/v2/view-builder/views/DeviceChallengePollView.js (hunk 3 above)
Comment (jhe-okta) [reply src/v2/view-builder/views/DeviceChallengePollView.js]: Dont worry this part yet. I just copied from some other view and forgot to remove it.

File: src/v2/view-builder/views/DeviceChallengePollView.js (hunk 4 above)
Comment (jhe-okta) [reply src/v2/view-builder/views/DeviceChallengePollView.js]: I thought we try to get rid of jquery?

**Comment by haishengwu-okta** [comment]
The work had been completed.

**Pull request closed without merging**